TA-puppet-alert-actions/bin/ta_puppet_alert_actions/puppet_plan_action.py

```diff
@@ -0,0 +1,130 @@
+
+  if alert['global']['timeout'] is not None and alert['global']['timeout'] is not '':
+    plan_timeout = alert['global']['timeout']
+  else:
+    plan_timeout = 600
+
+  try:
+    token_lifetime = int(plan_timeout) * 2
+  except Exception as e:
+    helper.log_error("Timeout must be an integer, '{}' was provided instead".format(plan_timeout))
+
+ # Check if the user is configured with an RBAC token or Password:
+  if alert['global']['pe_token']:
+    auth_token = bolt_user_pass
+  else:
+    auth_token = pie.rbac.genauthtoken(bolt_user,bolt_user_pass,'TA-puppet-alert-actions',rbac_url, timeout=token_lifetime)
+
+  # note: parameters is expected as a text string, not json, so in sample alert json must be represented as:
+  # "plan_parameters": "{ \"target\": \"puppet.angrydome.org\", \"service\": \"puppet\"}"
+
+  try:
+    jobid = pie.bolt.reqplan(plan_name,auth_token,puppet_environment,bolt_url,parameters=plan_parameters)
+    helper.log_debug('Puppet plan successfully requested with ID of {}'.format(jobid))
+  except Exception as e:
+    helper.log_error('Unable to request Puppet plan before {}'.format(e))
+
+  try:
+    jobresults = pie.bolt.getplanresult(jobid, auth_token, bolt_url, timeout=plan_timeout)
+  except Exception as e:
+    helper.log_error('Plan failed before {}'.format(e))
+
+  # While plans can be run against multiple targets the TargetSpec is typically a parameter named either 'target(s)' or 'nodes'. Although it can be named something entirely different.
+  # As such, we are currently reporting on the overall status of the Plan (i.e. If it succeeds on 2 nodes and fails on 1 it will be considered a 'failure').
+  # This also means that rmessage['message'] will include the PE Console hostname as opposed to each target the Plan was ran against.
+  rmessage = message
+  rmessage['action_state'] = jobresults['state']
+  rmessage['joburl'] = 'https://{}/#/plan_jobs/{}'.format(pe_console,jobid)
+  rmessage['pe_console'] = pe_console
+  rmessage['result'] = jobresults['result']
+  rmessage['start_timestamp'] = jobresults['created_timestamp']
+  rmessage['duration'] = jobresults['duration']
+  rmessage['finish_timestamp'] = jobresults['finished_timestamp']
+
+  if rmessage['action_state'] == 'success':
+    rmessage['message'] = 'Successfully ran plan {} on {} '.format(plan_name,pe_console)
+    helper.log_debug(rmessage['message'])
+  elif jobresults['state'] == 'failure':
+    rmessage['message'] = 'Failed to run plan {} on {} '.format(plan_name,pe_console)
+    helper.log_error(rmessage['message'])
+  else:
+    rmessage['message'] = 'Something happened to plan {} on {} that we have no idea about'.format(plan_name,pe_console)
+    helper.log_error(rmessage['message'])
+
+  pie.hec.post_action(rmessage, pe_console, splunk_hec_url, puppet_action_hec_token)
+
+# this is our interactive load option
+# assumes you're running this library directly from the command line
+# cat example_alert.json | python $thisfile.py
```

Comment:
this was probably copypastad from the example? - we should update the filenames to be accurate

Reply:
Good call, I went ahead and updated the steps for manually testing this.